sandbox/libs/authn/docs/jwt.md

```diff
@@ -37,7 +36,96 @@ There are 2 types of tokens:
 - Refresh Token - A Refresh Token is longer lasting and used to receive short-lived Access Tokens that actually confer access
 - Access Token - Access Tokens are short-lived tokens that confer access
 
-Creation of an asynchronous job or task should require a Refresh Token on creation which is associated with a user and let's the job run with the user's permissions. When the task
-starts execution it will use the refresh token to obtain an access token which will allow the task to interact with the cluster as the user who the token was created for.
+Creation of an asynchronous job or task should require a Refresh Token on creation which is associated with a subject and let's the job run with the subject's permissions. When the task
+starts execution it will use the refresh token to obtain an access token which will allow the task to interact with the cluster as the subject who the token was created for.
+
+Refresh tokens will be associated with a subject (and an extension?) and both the subject and administrators (or subjects with requisite permissions to revoke tokens of others) will be able to revoke access to the tokens. Issuing a token for oneself or on behalf of others will be another set of permissions that can be assigned.
+
+# Internal Cluster Actions
```

Comment:
Why treat internal actions differently from user actions?  This loosens the security model if we allow some actions to be trusted, its much cleaner if we don't have exceptions, what do you think?

Reply:
This was my attempt at documenting how internal actions are authorized today. I added this section in this document to show how it can possible with native security to authorize internal actions with the use of tokens and minimize the usage of the thread context.

Reply:
Maybe there is a way you could make it clearer that its from the old model and in the new model we will do X instead of Y?  Otherwise it it seems like the description of how we should build the feature forward

Reply:
Good suggestion, I will re-word this for the delegated authority proposal and make it sound less like my learnings about how the ThreadContext is shared across nodes and used for authorization with the security plugin.